taurus/entity/dict_serializable.py

```diff
@@ -82,7 +83,21 @@ def build(d):
         return loads(dumps(d))
 
     def __repr__(self):
-        return str(self.as_dict())
+        from taurus.util.impl import substitute_links, set_uuids
+        object_dict = self.as_dict()
+        # as_dict skips over keys in `skip`, but they should be in the representation.
+        skipped_keys = {x.lstrip('_') for x in vars(self) if x in self.skip}
+        for key in skipped_keys:
+            skipped_attribute = deepcopy(self.__getattribute__(key))
+            # Replace links in skipped keys with LinkByUID to prevent infinite recursion loop.
+            set_uuids(skipped_attribute)
```

Comment:
If we're setting uuids here in the copy, then they won't actually point back to the original object, right?

Reply:
Seems like if you called this twice, you'd see different ids each time.

Reply:
Well I don't think the copy would point to the wrong objects (or at least not the wrong ids), but if you made a link to the generated id here because you saw it in the output of a print statement, it seems like it would be an invalid one.

Reply:
The copy only lives for the lifetime of the `__repr__` function afaict

Reply:
The uid in this case is meaningless, because the actual object doesn't know about it.

Reply:
We could make it a real action and actually set the uids.